specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,313 @@
+<!-- TEMPLATE
+  The purpose of this spec is to describe a new WebView2 feature and its APIs.
+
+  There are two audiences for the spec. The first are people
+  that want to evaluate and give feedback on the API, as part of
+  the submission process. When it's complete
+  it will be incorporated into the public documentation at
+  docs.microsoft.com (https://docs.microsoft.com/en-us/microsoft-edge/webview2/).
+  Hopefully we'll be able to copy it mostly verbatim.
+  So the second audience is everyone that reads there to learn how
+  and why to use this API. 
+-->
+
+
+# Background
+These APIs will allow users to drop things such as images, text, and links into the WebView as part of a drag/drop operation.
+The reason that we need separate APIs for this with composition hosting is because we don't have an HWND to call RegisterDragDrop on.
+The hosting app needs to call RegisterDragDrop on the HWND that contains the WebView and implement IDropTarget so it can forward the
+calls (IDropTarget::DragEnter, DragMove, DragLeave, and Drop) to the WebView. Other UI frameworks have their own ways to register.
+For example, Xaml require setting event handler for DragEnter, DragOver, DragLeave, and Drop on the corresponding UIElement.
+
+Additionally, for win32, the hosting app also needs to call into IDropTargetHelper before forwarding those calls to us as documented here:
+https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-idroptargethelper
+
+For API reviewers, We want a unified API surface between COM and WinRT that works in both UWP and Win32.
+
+We could:
+1. Have one API focusing on Win32 types and require the end dev to convert from UWP types to Win32 (which we've done).
+2. Have one API focusing on UWP types and require the end dev to convert Win32 to UWP.
+3. Have two sets of methods one for Win32 and one for UWP types.
+Or we could do (1) or (2) and provide a conversion function.
+Because the conversion is simple and we have a large Win32 user base we chose (1) and provided a conversion function.
+
+
+# Description
+DragEnter, DragOver, DragLeave, and Drop are functions meant to provide a way for composition hosted WebViews to receive drop events as part of a drag/drop operation.
+It is the hosting application's responsibility to call RegisterDragDrop (https://docs.microsoft.com/en-us/windows/win32/api/ole2/nf-ole2-registerdragdrop)
+on the HWND that contains any composition hosted WebViews and to implement IDropTarget(https://docs.microsoft.com/en-us/windows/win32/api/oleidl/nn-oleidl-idroptarget)
```

Comment:
Does it have to register on the immediate host of the WebView, or can it register on the top-level parent?